### PR TITLE
Dependency & Node/NPM version fixes to package.json

### PR DESCRIPTION
**Pull request by adammessinger**
- Depend on `grunt` v0.4.x in package.json, not `grunt-cli`. This is the approach recommended by the Grunt docs, and avoids `peerDependencies` failures when dependencies try to install Grunt 1.0.x.
- State Node & NPM compatibility in package.json. Link-SF will not install properly with Node > 0.12.x or NPM 3+.

### CODE REVIEW

**Comment by jden** [comment]
Thanks for opening the PR! Please know that we're currently doing active development on the `open-referral` branch, which may have already updated the dev tooling. /cc @thebritican

**Comment by adammessinger** [comment]
I understand that, but it doesn't do much to help people that would like to put the stable `master` branch to work in their community today. A local project I'm working on switched to using [ohana-api](https://github.com/codeforamerica/ohana-api) and [ohana-web-search](https://github.com/codeforamerica/ohana-web-search) because the work required to get link-sf up and running was just too much of a time-suck. It was frustrating, since I really like this project's simple UI design and the static site approach it's taken.

Another thing your team might consider: using [`npm shrinkwrap`](https://docs.npmjs.com/cli/shrinkwrap), or even [`shrinkpack`](https://www.npmjs.com/package/shrinkpack), to lock down dependencies and make installation of the `master` branch more reliable. Also, the docs could use updates to cover working with the open-source Parse Server. It's no longer possible to setup a new Parse.com account like the current instructions tell you to.

**Comment by thebritican** [comment]
Hey @adammessinger I'm really sorry you had this rough experience getting set up on `linksf`. The dev tooling is certainly outdated on `master`. We meet every Wednesday, so we'll have a discussion about making time to improve the Parse version's dev experience. We're on tight timeline currently, here's some background:

Our response to Facebook's announcement of discontinuing parse was to begin immediate development enlisting a new **BaaS**. We're attempting to finish this non Parse version within the next month. The new version will also be more compatible with the ohana api, since the data will be presented according to the Open Referral spec.

**TL;DR** we want to fix this, but need to see if we can dedicate time before the non Parse release.

**Comment by adammessinger** [comment]
I understand where you're coming from wrt time pressures and resource constraints -- I've been in similar situations. I'd pitch in more than just this one pull request for your project, but my energies right now are focused on my own resource site project. It's just become a "focus project" for Open Wichita, which is likely to pick up the pace.

I had an architectural thought that might save Link-SF some trouble in the future. You could decide that some standard API, like Ohana, is the way that Link-SF will interact with any back-end. Then write an adapter for your preferred BaaS (Firebase) and accept submission of other adapters from users. Those other adapters might not have any official support, but could still be offered up on GitHub for anyone that wants to employ Link-SF to learn from and adjust to suit their needs. It could be as simple as reviewing submissions and adding a repo link for good ones to a page in the Link-SF wiki. It could be the start of your own small ecosystem that helps Link-SF adapt more nimbly to future disruptions like the Parse closure.

**Comment by jden** [comment]
@adammessinger that's not a bad idea! As a former Code for America fellow myself, I know it can be a challenge walking the line between the ideal technical solution and a "good enough" approach that will help get software out of github and into the community, where it can actually do some good.

Would it help you in the immediate term if we merged this PR onto master as-is, regardless of our future plans that we talked about above?

**Comment by adammessinger** [comment]
It wouldn't help my project, but it might help someone else get Link-SF up and running for their community faster. Right now, Open Wichita is focused on getting the Ohana-based solution customized for our city and populated with relevant resource info.

In the longer run, I could see Link-SF working well as a light-weight and mobile-friendly static website for quick access to the kinds of resources that people most in need want to find quickly and easily. That's how I imagined putting it to work in Wichita, perhaps employing Ohana to get info about a broader variety of resources to a wider audience later. Now we're working in the other direction, and I could still see Link-SF being useful in Wichita (and many other places). Just not right this moment, in our case. :-)

**Comment by jden** [comment]
Cool, thanks for the feedback! I think that sounds in line with what we're working towards.

**Comment by craig-day** [comment]
I'm going to close this PR because the React version of the application has been merged to master and deployed in production.
